prov/tcp/src/tcpx_cq.c

```diff
@@ -64,25 +73,25 @@ struct tcpx_xfer_entry *tcpx_xfer_entry_alloc(struct tcpx_cq *tcpx_cq)
 		return NULL;
 	}
 
-	xfer_entry = util_buf_alloc(tcpx_cq->xfer_entry_pool);
+	xfer_entry = util_buf_alloc(tcpx_cq->buf_pools[type].pool);
 	if (!xfer_entry) {
 		tcpx_cq->util_cq.cq_fastlock_release(&tcpx_cq->util_cq.cq_lock);
 		FI_INFO(&tcpx_prov, FI_LOG_DOMAIN,"failed to get buffer\n");
 		return NULL;
 	}
 	tcpx_cq->util_cq.cq_fastlock_release(&tcpx_cq->util_cq.cq_lock);
-	memset(xfer_entry, 0, sizeof(*xfer_entry));
 	return xfer_entry;
 }
 
 void tcpx_xfer_entry_release(struct tcpx_cq *tcpx_cq,
-			   struct tcpx_xfer_entry *xfer_entry)
+			     struct tcpx_xfer_entry *xfer_entry)
 {
 	if (xfer_entry->ep->cur_rx_entry == xfer_entry)
 		xfer_entry->ep->cur_rx_entry = NULL;
```

Comment:
Unrelated to this change set, the clearing of cur_rx_entry looks racy.  Is there another lock being held here?

Reply:
cur_rx_entry is protected by queue_lock which is held when function is called.